src/Workspaces/Core/Portable/Storage/AbstractPersistentStorageService.cs

```diff
@@ -249,6 +254,15 @@ public void Dispose()
                 _storage.Dispose();
             }
 
+            public Task<Checksum> ReadChecksumAsync(string name, CancellationToken cancellationToken = default)
+                => _storage.Target.ReadChecksumAsync(name, cancellationToken);
+
+            public Task<Checksum> ReadChecksumAsync(Project project, string name, CancellationToken cancellationToken = default)
+                => _storage.Target.ReadChecksumAsync(project, name, cancellationToken);
```

Comment:
I forgot how thie ReferenceCountDisposableWrapper works. so is it safe to just assume Target exist? or are we assume that it will always exist until VS shutdown? what guarantee that this won't get called once VS shutdown process start (I mean after _storage.Dispose is called) due to async work?

Reply:
@sharwell .  I don't know how this part of the system works that well.